ICAFIX/hcp_fix_multi_run

```diff
@@ -838,86 +872,156 @@ fi
+		if [[ ! -f "${TrainingData}" ]]; then
+			log_Err_Abort "FIX training data not found: ${TrainingData}"
+		fi
+	else
+		# User has not specified a training data file
+		log_Msg "training data file not specified"
+		TrainingData="${FSL_FIXDIR}"/training_files/HCP_Style_Single_Multirun_Dedrift.RData
+		if [[ ! -f "$TrainingData" ]]; then
+			log_Err_Abort "default training file '$TrainingData' not found, please update your fix installation"
+		fi
 	fi
-	
-	# if the specified TrainingData is not a full path to an existing file,
-	# assume that the user is specifying the name of a file in the training_files folder in FSL_FIXDIR
-	if [ ! -f "${TrainingData}" ]; then 
-		TrainingData=${FSL_FIXDIR}/training_files/${TrainingData}
+
+	## MPH: The 'fix' script itself will continue to log to its own custom files
+	## Alert user to where those are
+	log_Msg "Check ${concatfmrihp}.ica/fix/logMatlab.txt for log output from feature extraction"
+	log_Msg "Check ${concatfmrihp}.ica/.fix_2b_predict.log for log output from component classification"
+	log_Msg "Check ${concatfmrihp}.ica/.fix.log for log output from cleanup stage"
+
+else
+	log_Msg "Running pyFIX"
+
+	# Get concatenated Movement_Regressors_demean.txt into the format expected by pyfix
+	if [ -f Movement_Regressors_demean.txt ] ; then
+		log_Debug_Msg "About to create ${concatfmrihp}.ica/mc/prefiltered_func_data_mcf.par file"
+		cat Movement_Regressors_demean.txt | awk '{ print $4 " " $5 " " $6 " " $1 " " $2 " " $3}' > ${concatfmrihp}.ica/mc/prefiltered_func_data_mcf.par
+	else
+		log_Err_Abort "Movement_Regressors.txt not retrieved properly." 
 	fi
-	
-	# finally, if the TrainingData file is not found, report an error and get out of here
-	if [[ ! -f "${TrainingData}" ]]; then
-		log_Err_Abort "FIX training data not found: ${TrainingData}"
+
+	# if specified, copy template brain for creation of masks
+	if [[ ${T1wTemplateBrain} ]]; then
+		$FSLDIR/bin/imcp ${T1wTemplateBrain} ${concatfmrihp}.ica/reg/standard
 	fi
-else
-	# User has not specified a training data file
-	log_Msg "training data file not specified"
-	TrainingData="${FSL_FIXDIR}"/training_files/HCP_Style_Single_Multirun_Dedrift.RData
-	if [[ ! -f "$TrainingData" ]]; then
-		log_Err_Abort "default training file '$TrainingData' not found, please update your fix installation"
+
+	# Handle user specified training data file
+	if [[ "${TrainingData}" != "" ]]; then
+		# User has specified a training data file
+		
+		# if training data is in legacy RData format, and not a built-in model, convert it
+		builtInModels="@HCP25_hp2000@HCP7T_hp2000@HCP_Style_Single_Multirun_Dedrift@NHP_HCP_Macaque@NHP_HCP_MacaqueCyno@Standard@UKBiobank@WhII_MB6@WhII_Standard@"
+		if [[ "${TrainingData}" == *.RData ]]; then 
+			TrainingData=${TrainingData%.RData}
+			if [[ ! ${builtInModels} == *"@${TrainingData}@"* ]]; then 
+				${FSL_FIXDIR}/convert_fix_model ${TrainingData}.RData ${TrainingData}.pyfix_model
```

Comment:
Is the user guaranteed to have write access to the output location of the `.pyfix_model` generated by this `convert_fix_model` command?  Also, above for the `legacyFix` condition, we have code to try to locate the `TrainingData` file, if the user didn't provide the full path (or appropriate relative path for whatever is the working directory at this point in the code).  That helpful assistance seems to have been dropped here.

Reply:
I think the working directory is next to the .ica folder when this is done, so the folder should be writable (there are `mkdir`s without `-p` that rely on it).  I don't know the exact logic of convert_fix_model's output location, though.

As for looking for it in other places, this code is only hit when pyfix is being used.  I suppose we could add the logic.

Reply:
Actually, I don't know if we can add the logic, because we would need to know the install location of legacy fix.  The current code clobbers a legacy `FSL_FIXDIR` in the pyfix case and then uses it to run pyfix commands.  It also appears to allow using a different copy of pyfix than the one in `FSL_DIR` by setting the same env variable, so that is fun...

Reply:
Perhaps we can address this one after the fact?  The solution is not clear to me either.